src/AWS/PubKeys.hs

```diff
@@ -35,16 +36,28 @@ import Text.Printf (printf)
 import Prelude hiding (filter, takeWhile)
 
 
-getPubKeys :: [Ec2Instance] -> IO [Ec2Instance]
-getPubKeys = fmap catMaybes <$> mapConcurrently getPubKey
+getPubKeys :: [Ec2Instance] -> IO [Either Ec2Instance Ec2Instance]
```

Comment:
why `Either Ec2Instance Ec2Instance`?

Reply:
Left is failure, right is success.

You get all of the data associated with the one that failed in the case it failed.